feat(curriculum): add string method workshop (#66936)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-string-method/69de4308e652aaeb5ccdb03e.md

---
id: 69de4308e652aaeb5ccdb03e
title: Step 1
challengeType: 20
dashedName: step-1
---

# --description--

In this workshop, you will learn how to work with strings in Python using different string methods.

Strings are used to store text data like `names`, `emails`, or `addresses`. For example:

```py
email = '' # empty string
```

An empty string is a string that has no characters inside the quotes.

In this step, you will create a variable and assign it an empty string.

> Create a variable named `email` and assign it an empty string.

Reviewer: There's already a workshop about strings before this one. So I would merge step 1 and step 2:
```suggestion
In this workshop, you will practice working with Python's string methods.

Start by creating a variable named `email` and assign it the string `alice.johnson@company.com`.
```

Reviewer: Please apply this with all the related changes about tests and following step

> Create a variable named `email` and assign it an `empty string`.

Reviewer 2: ```suggestion
Create a variable named `email` and assign it an empty string.
```

# --hints--

You should define a variable called `email`.

```js
({ test: () => assert(runPython(`_Node(_code).has_variable("email")`)) })
```

> You should assign a empty `string` value to `email`.

Reviewer 2: ```suggestion
You should assign a empty string value to `email`.
```

---
id: 69de4308e652aaeb5ccdb03e
title: Step 1
challengeType: 20
dashedName: step-1
# demoType can either be 'onClick' or 'onLoad'. If the project or lab doesn't have a preview, delete the property
# demoType: onClick
---

# --description--

In this workshop, you will learn how to work with strings in Python using different string methods.

Strings are used to store text data like `names`, `emails`, or `addresses`. For example:

```python
name = "" #empty string
```

An empty string is a string that has no characters inside the quotes.

In this step, you will create a variable and assign it an empty string.

Create a variable named `email` and assign it an `empty string`.

# --hints--

You should define a variable called `email`.

```js
({ test: () => assert(runPython(`_Node(_code).has_variable("email")`)) })
```

You should assign a empty `string` value to `email`.

```js
({ test: () => assert(runPython(`
node = _Node(_code).find_variable("email")
not node.is_empty() and (
  node.is_equivalent("email = ''") or 

>   node.is_equivalent('email = ""')

Reviewer 2: `email = ''` and `email = ""` are the same node, you don't need to check for both
